hi-ml/src/health_ml/networks/layers/attention_layers.py

```diff
@@ -24,6 +24,21 @@ def forward(self, features: Tensor) -> Tuple[Tensor, Tensor]:
         return (attention_weights, pooled_features)
 
 
+class MaxPoolingLayer(nn.Module):
+    """Max pooling returns uniform weights and the maximum feature vector over the first axis"""
+
+    # args/kwargs added here for compatibility with parametrised pooling modules
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__()
+
+    def forward(self, features: Tensor) -> Tuple[Tensor, Tensor]:
+        num_instances = features.shape[0]
+        attention_weights = torch.full((1, num_instances), 1. / num_instances)
```

Comment:
[Speculative] I understand there's no obvious way to assign instance attentions when the `max` is taken dimension-wise. Do you think it could it make sense to extract a normalised 'histogram' of selected indices? I.e. the attention weight of each instance being proportional to the number of dimensions for which its value was selected (e.g. using `torch.bincount(indices, minlength=num_dimensions)`). Might this be useful as a proxy?

Reply:
In any case, it would be helpful to add a comment here to clarify that the weights currently returned are arbitrarily uniform because max-pooling works per dimension.

Reply:
It is a good idea to add weights based on how many times the maximum value was selected. I have changed it in the code, please have a look